src/parse_execution.cpp

```diff
@@ -1445,6 +1493,57 @@ parse_execution_result_t parse_execution_context_t::eval_node(tnode_t<g::stateme
     return status;
 }
 
+parse_execution_result_t parse_execution_context_t::run_variable_assignment_prefixed_statement(
+    tnode_t<grammar::variable_assignment_prefixed_statement> var_statement,
+    const block_t *associated_block) {
+    tnode_t<g::variable_assignment_list> var_list_node = var_statement.child<0>();
+    tnode_t<g::statement> wrapped_statement = var_statement.child<1>();
+    variable_assignment_node_list_t var_list = get_variable_assignment_nodes(var_list_node);
+    const block_t *block = nullptr;
+    cleanup_t pop_block([&]() {
+        if (block) parser->pop_block(block);
+    });
+    if (!var_list.empty()) {
+        block = parser->push_block(block_t::variable_assignment_block());
+    }
+    for (const auto &variable_assignment : var_list) {
+        const wcstring &source = variable_assignment.get_source(pstree->src);
+        auto equals_pos = variable_assignment_equals_pos(source);
+        assert(equals_pos != wcstring::npos);
+        const wcstring &variable_name = source.substr(0, equals_pos);
```

Comment:
Unfortunately we can't take a reference here as it is a returned temporary, this should be `wcstring variable_name =` and `wcstring expression =`